interpreter/core/llm/llm.py

```diff
@@ -166,17 +166,24 @@ def run(self, messages):
         ## Start forming the request
 
         params = {
-            "model": self.model,
             "messages": messages,
             "stream": True,
         }
 
+        if self.model:
+            # if model is azure, set the deployment_id and model
+            if self.model.startswith("azure/"):
+                params["deployment_id"] = self.model.split("/").pop()
+                params["model"] = params["deployment_id"]
+            # otherwise, set only the model
+            else:
+                params["model"] = self.model
+
         # Optional inputs
         if self.api_key:
             params["api_key"] = self.api_key
         if self.api_base:
             params["api_base"] = self.api_base
-            params["custom_llm_provider"] = "openai"
```

Comment:
I've opted to remove the line params["custom_llm_provider"] = "openai" from the code. 
This adjustment is based on the understanding that the custom_llm_provider parameter is meant for specifying non-OpenAI language model providers, which isn't necessary when utilizing OpenAI models. 

https://github.com/BerriAI/litellm/blob/58fb6b2b5924041880b9aca1a7f653653a3527a4/litellm/main.py#L440